### PR TITLE
Add Kafka OTEL receiver/ingester

### PR DESCRIPTION
**Pull request by pavolloffay**
Related to https://github.com/jaegertracing/jaeger/issues/2205 

This PR adds Kafka OTEL receiver. This receiver will be used in OTEL ingester that will come in a follow-up PR.

PR for kafka exporter https://github.com/jaegertracing/jaeger/pull/2135

### CODE REVIEW

**Comment by pavolloffay** [comment]
@rubenvp8510 or and @objectiser could you please review?

File: cmd/opentelemetry-collector/app/receiver/kafka/config_test.go [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+	}}
+
+	factories.Receivers[TypeStr] = factory
+	cfg, err := config.LoadConfigFile(t, path.Join(".", "testdata", "config.yaml"), factories)
+	require.NoError(t, err)
+	require.NotNil(t, cfg)
+
+	kafkaCfg := cfg.Receivers[TypeStr].(*Config)
+	assert.Equal(t, TypeStr, kafkaCfg.Name())
+	assert.Equal(t, "jaeger-prod", kafkaCfg.Topic)
```
Comment (objectiser) [inline cmd/opentelemetry-collector/app/receiver/kafka/config_test.go:77]: Might be good to allow one of the flag values to be passed to the final config?

File: cmd/opentelemetry-collector/app/receiver/kafka/config_test.go [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+	assert.Equal(t, "emojis", kafkaCfg.Encoding)
+	assert.Equal(t, []string{"foo", "bar"}, kafkaCfg.Options.Brokers)
+	assert.Equal(t, "tls", kafkaCfg.Options.Authentication)
+	assert.Equal(t, "user", kafkaCfg.Options.PlainText.UserName)
+	assert.Equal(t, "123", kafkaCfg.Options.PlainText.Password)
+	assert.Equal(t, true, kafkaCfg.Options.TLS.Enabled)
+	assert.Equal(t, "key.crt", kafkaCfg.Options.TLS.KeyPath)
```
Comment (objectiser) [inline cmd/opentelemetry-collector/app/receiver/kafka/config_test.go:84]: Need to check `ca.crt` aswell.

**Comment by pavolloffay** [comment]
> Main question, not for this PR, is how we work with/transition to a pure OTel/Kafka impl - would be better if using OTel model for kafka messages.

Yeah, this is not related to this PR, but it is definitely something on the roadmap.

File: cmd/opentelemetry-collector/app/receiver/kafka/config_test.go (hunk 1 above)
Comment (pavolloffay) [reply cmd/opentelemetry-collector/app/receiver/kafka/config_test.go:77]: `kafkaCfg.Options.Kerberos.Username` is passed from jaeger config file

File: cmd/opentelemetry-collector/app/defaults/defaults.go [hunk 3]
```diff
@@ -25,7 +25,7 @@ import (
 	"github.com/spf13/pflag"
 	"github.com/spf13/viper"
 
-	"github.com/jaegertracing/jaeger/cmd/ingester/app"
+	ingestrApp "github.com/jaegertracing/jaeger/cmd/ingester/app"
```
Comment (objectiser) [inline cmd/opentelemetry-collector/app/defaults/defaults.go:28]: `ingesterApp`

File: cmd/opentelemetry-collector/app/receiver/kafka/config_test.go (hunk 1 above)
Comment (objectiser) [reply cmd/opentelemetry-collector/app/receiver/kafka/config_test.go:77]: The issue is that none of the CLI flag values are making it into the final config - would be good if atleast one did for completeness.

Comment (pavolloffay) [reply cmd/opentelemetry-collector/app/receiver/kafka/config_test.go:77]: Flags or config file are the same source for viper. A couple of lines above we test that flags are passed to the factory.

**Pull request merged**